include/seqan3/std/concept/core_language.hpp

```diff
@@ -90,11 +92,17 @@ concept bool common_reference_concept =             static_cast<bool>(ranges::Co
 template <typename t, typename u, typename ...rest>
 concept bool common_concept =                       static_cast<bool>(ranges::Common<t, u, rest...>());
 
+/*!\brief Resolves to `std::is_arithmetic_v<type>`.
+ * \sa https://en.cppreference.com/w/cpp/types/is_arithmetic
+ */
+template <typename t>
+concept bool arithmetic_concept =                   std::is_arithmetic_v<t>;
+
 /*!\brief Resolves to `ranges::Integral<type>()`
  * \sa http://en.cppreference.com/w/cpp/experimental/ranges/concepts/Integral
  */
 template <typename t>
-concept bool integral_concept =                     static_cast<bool>(ranges::Integral<t>());
+concept bool integral_concept =                     arithmetic_concept<t> && static_cast<bool>(ranges::Integral<t>());
```

Comment:
why not using `std::is_integral_v`?

Reply:
It would produce a better error message, as the constraint failure is not somewhere within the ranges library.

Reply:
There is a bunch of concepts that are not yet implemented natively and not documented properly. I didn't want to do clean-up in the same PR as features.

Reply:
Ah ok, I missed, that the integral_concept was just extended.